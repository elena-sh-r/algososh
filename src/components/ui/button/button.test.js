import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen, fireEvent } from '@testing-library/react';

import { Button } from './button';

const buttonText = 'Какой-то текст кнопки';

describe('Тестирование компонента Button', () => {
  it('Кнопка с текстом отрисовывается корректно', () => {
    const button = renderer
      .create(<Button text={buttonText} />)
      .toJSON();
      expect(button).toMatchSnapshot();
  }); 
  
  it('Кнопка без текста отрисовывается корректно', () => {
    const button = renderer
      .create(<Button />)
      .toJSON();
      expect(button).toMatchSnapshot();
  }); 
  
  it('Заблокированная кнопка отрисовывается корректно', () => {
    const button = renderer
      .create(<Button disabled={true} />)
      .toJSON();
      expect(button).toMatchSnapshot();
  }); 
  
  it('Кнопка с индикацией загрузки отрисовывается корректно', () => {
    const button = renderer
      .create(<Button isLoader={true} />)
      .toJSON();
      expect(button).toMatchSnapshot();
  }); 
});

describe('Тестирование вызова колбека при клике на кнопку', () => {
  it('Корректность вызова колбека при клике на кнопку', () => {
    const fn = jest.fn();
    render(<Button text={buttonText} onClick={fn} />)
    const button = screen.getByText(buttonText);
    fireEvent.click(button);
        
    expect(fn).toHaveBeenCalledTimes(1);
  });
});
