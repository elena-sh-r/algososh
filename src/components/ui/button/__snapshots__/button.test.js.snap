// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Заблокированная кнопка отрисовывается корректно 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Кнопка без текста отрисовывается корректно 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Кнопка с индикацией загрузки отрисовывается корректно 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Кнопка с текстом отрисовывается корректно 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Какой-то текст
  </p>
</button>
`;

exports[`Тестирование компонента Button Заблокированная кнопка отрисовывается корректно 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Тестирование компонента Button Кнопка без текста отрисовывается корректно 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Тестирование компонента Button Кнопка с индикацией загрузки отрисовывается корректно 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Тестирование компонента Button Кнопка с текстом отрисовывается корректно 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Какой-то текст кнопки
  </p>
</button>
`;
